app/src/main/java/com/bnyro/clock/ui/model/TimerModel.kt

```diff
@@ -120,52 +140,34 @@ class TimerModel : ViewModel() {
         if (scheduledObjects.isEmpty()) context.unbindService(serviceConnection)
     }
 
-    fun addNumber(number: String) {
-        // Adding 0 to 0 makes no sense
-        if (number.toIntOrNull() == 0 && timePickerSecondsState == "0") return
-
-        // Couldn't find a better way to substring
-        timePickerSecondsState = (timePickerSecondsState + number)
-            .padEnd(7, 'x')
-            .substring(0, 7)
-            .replace("x", "")
-    }
-
-    fun addSeconds(seconds: Int) {
-        timePickerSecondsState = getSecondsStringPadded().substring(0, 4) +
-            seconds.toString().padStart(2, '0')
-    }
-
-    fun addMinutes(minutes: Int) {
-        timePickerSecondsState = getSecondsStringPadded().substring(0, 2) +
-            minutes.toString().padStart(2, '0') +
-            getSecondsStringPadded().substring(4)
-    }
+    /* =============== Numpad time picker ======================== */
+    var timePickerFakeUnits by mutableStateOf(
+        0,
+        policy = object : SnapshotMutationPolicy<Int> {
+            override fun equivalent(a: Int, b: Int): Boolean {
+                if (a == b) return true
+                b.let {
+                    val roughHours = it.div(10000).mod(100)
+                    val roughMinutes = it.div(100).mod(100)
+                    val roughSeconds = it.mod(100)
+                    timePickerSeconds =
+                        roughSeconds + roughMinutes.times(60) + roughHours.times(3600)
+                }
```

Comment:
```suggestion
                    val roughHours = (b / 10000) % 100
                    val roughMinutes = (b / 100) % 100
                    val roughSeconds = b % 100
                    timePickerSeconds =
                        roughSeconds + roughMinutes * 60 + roughHours * 60
```
I don't like doing math with extension functions personally, because it's likely just less performant and less readable.

Reply:
1. `div()` and `/` are both the same thing. There's no performance benefit from using one over other (https://kotlinlang.org/docs/operator-overloading.html#arithmetic-operators)

2. I think using `div`, `mod` is more readable because it feels like reading a natural language. Also it helps reduce the use of brackets. Using brackets is more confusing and prone to mistakes IMO.

Reply:
1. Fair enough.
2. All common programming languages use normal math expressions for calculations, and nothing text based.
I think it's generally just easier to read when you have math expressions in front of you like you're familiar with from calculating with pen and paper. It certainly can reduce the usage of nesting with brackets, but we have only one level of brackets at most, so I don't think that's reasonable for our use case.